Marlin/src/module/temperature.cpp

```diff
@@ -499,8 +499,9 @@ volatile bool Temperature::raw_temps_ready = false;
    * Needs sufficient heater power to make some overshoot at target
    * temperature to succeed.
    */
-  void Temperature::PID_autotune(const_float_t target, const heater_id_t heater_id, const int8_t ncycles, const bool set_result/*=false*/) {
+  void Temperature::PID_autotune(const celsius_t desired_target, const heater_id_t heater_id, const int8_t ncycles, const bool set_result/*=false*/) {
     float current_temp = 0.0;
+    float target = (float)desired_target;
```

Comment:
Does the `explicit` keyword makes this explicit cast necessary? I hope this is not needed. Call me old fashioned, but I feel that it could annoy neophyte contributors when the language appears to be obtuse, coy, or confused about the type of the lvalue, and then they will come around asking for help and use up developer time.

Reply:
I understand your concern here. Yes `explicit` requires an explicit cast here. I see it as an advantage here so we can spot each time we query the conversion operator (which is not zero cost, like for casting a int to short or something). But if you feel it's too much verbose, once all the changes are applied and validated, I'll remove the explicit conversion operator so it does not appear everywhere in the code base in a single commit.